test/lib/users/ssh_authorized_keyring_test.rb

```diff
@@ -90,122 +130,163 @@ def authorized_keys_from_home(path)
+        let(:key) { keyring.keys.first }
 
-      it "SSH directory and authorized_keys inherits owner/group from home" do
-        allow(Yast::FileUtils).to receive(:GetOwnerUserID).with(home).and_return(uid)
-        allow(Yast::FileUtils).to receive(:GetOwnerGroupID).with(home).and_return(gid)
-        expect(Yast::FileUtils).to receive(:Chown).with("#{uid}:#{gid}", ssh_dir, false)
-        expect(Yast::FileUtils).to receive(:Chown)
-          .with("#{uid}:#{gid}", authorized_keys_path, false)
+        it "does not write keys again" do
+          expect(file).to_not receive(:save)
 
-        keyring.write_keys
+          keyring.write_keys
+        end
       end
 
-      it "sets SSH directory permissions to 0700" do
-        keyring.write_keys
-        mode = File.stat(ssh_dir).mode.to_s(8)
-        expect(mode).to eq("40700")
-      end
+      context "but new keys are added" do
+        it "writes the keys" do
+          expect(file).to receive(:save)
 
-      it "sets authorized_keys permissions to 0600" do
-        keyring.write_keys
-        mode = File.stat(authorized_keys_path).mode.to_s(8)
-        expect(mode).to eq("100600")
-      end
+          keyring.write_keys
+
+          expect(File).to exist(authorized_keys_path)
+        end
 
-      context "when home directory does not exist" do
-        let(:home_dir_exists) { false }
+        it "SSH directory and authorized_keys inherits owner/group from home" do
+          allow(Yast::FileUtils).to receive(:GetOwnerUserID).with(home).and_return(uid)
+          allow(Yast::FileUtils).to receive(:GetOwnerGroupID).with(home).and_return(gid)
+          expect(Yast::FileUtils).to receive(:Chown).with("#{uid}:#{gid}", ssh_dir, false)
+          expect(Yast::FileUtils).to receive(:Chown)
+            .with("#{uid}:#{gid}", authorized_keys_path, false)
 
-        it "raises a HomeDoesNotExist exception and does not write authorized_keys" do
-          expect(Yast::Users::SSHAuthorizedKeysFile).to_not receive(:new)
-          expect { keyring.write_keys }
-            .to raise_error(Yast::Users::SSHAuthorizedKeyring::HomeDoesNotExist)
+          keyring.write_keys
         end
-      end
 
-      context "when SSH directory could not be created" do
-        it "raises a CouldNotCreateSSHDirectory exception and does not write authorized_keys" do
-          expect(Yast::Users::SSHAuthorizedKeysFile).to_not receive(:new)
-          expect(Yast::SCR).to receive(:Execute)
-            .with(Yast::Path.new(".target.mkdir"), anything)
-            .and_return(false)
-          expect { keyring.write_keys }
-            .to raise_error(Yast::Users::SSHAuthorizedKeyring::CouldNotCreateSSHDirectory)
+        context "when home directory does not exist" do
+          let(:home_dir_exists) { false }
+
+          it "raises a HomeDoesNotExist exception and does not write authorized_keys" do
+            expect(Yast::Users::SSHAuthorizedKeysFile).to_not receive(:new)
```

Comment:
why expect on new and not on save?

Reply:
This test was already there, just moved around. Not sure about it.